config/auth_databricks_oidc.go

```diff
@@ -0,0 +1,94 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	"github.com/databricks/databricks-sdk-go/config/experimental/auth"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Creates a new Databricks OIDC TokenSource.
+func NewDatabricksOIDCTokenSource(
+	cfg *DatabricksOIDCTokenSourceConfig,
+) auth.TokenSource {
```

Comment:
[nit] Go does not enforce a specific line length; prefer this format when there is a long list of parameters

```suggestion
func NewDatabricksOIDCTokenSource(cfg *DatabricksOIDCTokenSourceConfig) auth.TokenSource {
```